### PR TITLE
Enable warning flag -Wshorten-64-to-32

### PR DESCRIPTION
**Pull request by chaupow**
# Issue

This PR works towards #3164 and starts with solving all warnings that come with enabling `-Wshorten-64-to-32`

cc @springmeyer 
## Tasklist
- [ ] KILL ALL WARNINGS
- [ ] add `-Wshorten-64-to-32` and `-Werror`(?) to CMake File 
- [ ] rebase with master
- [ ] review
- [ ] adjust for comments

### CODE REVIEW

File: src/extractor/compressed_edge_container.cpp [hunk 1]
```diff
@@ -76,8 +77,8 @@ unsigned CompressedEdgeContainer::GetZippedPositionForReverseID(const EdgeID edg
 void CompressedEdgeContainer::SerializeInternalVector(const std::string &path) const
 {
     boost::filesystem::fstream geometry_out_stream(path, std::ios::binary | std::ios::out);
-    const unsigned compressed_geometry_indices = m_compressed_geometry_index.size() + 1;
-    const unsigned compressed_geometries = m_compressed_geometry_nodes.size();
+    const std::size_t compressed_geometry_indices = m_compressed_geometry_index.size() + 1;
+    const std::size_t compressed_geometries = m_compressed_geometry_nodes.size();
```
Comment (daniel-j-h) [inline src/extractor/compressed_edge_container.cpp]: For all sizes written to disk it makes sense to use fixed-size integral types.

**Comment by daniel-j-h** [comment]
`Werror` is problematic for multiple reasons, one is that we support a broad range of compilers of which some are strict and some are stricter. For example, gcc 4.9 issues warnings you won't see on Travis with more recent compilers (since they're wrong). Making these builds fail hard is not an option.

**Comment by springmeyer** [comment]
@daniel-j-h I agree with you that `-Werror` would be problematic now and is not realistic short-term, but I think it should be a goal. Both `mbgl` and `v8` use `-Werror` by default because they are able to build with minimal warnings across a large range of compilers.

**Comment by springmeyer** [comment]
interesting read in this context of this work: http://blog.reverberate.org/2012/12/testing-for-integer-overflow-in-c-and-c.html

**Comment by daniel-j-h** [comment]
What's the state of this? Asking since @MoKob did a PR today fixing the `int` vs `size_t` differences, too. Want to rebase against master, squash changes and then already merge this into master?

**Comment by springmeyer** [comment]
See also https://github.com/Project-OSRM/osrm-backend/issues/3483

**Comment by daniel-j-h** [comment]
Ping - what's the state of this? We're having this PR around now for three month; it's not up to date or rebased on master and the contractor size\_t issues already fixed by @MoKob. If it's not a priority at the moment please at least rebase and let's get the changes in - otherwise I'm inclined to close this.

**Comment by daniel-j-h** [comment]
Same with https://github.com/Project-OSRM/osrm-backend/issues/3247 - if it's a part of this ticket and / or stale please update accordingly.

**Comment by chaupow** [comment]
@springmeyer I don't have to time to work on this atm 😞 
Should we just close this for now?

**Comment by springmeyer** [comment]
Let's close. I propose the next action be that we re-issue a new PR that actually enables this warning. Then we can fix the warnings in followup PRs (and look back at this one for hints).

**Comment by springmeyer** [comment]
See also #4495

**Pull request closed without merging**